File: mySolutions/240. Search a 2D Matrix II/240. Search a 2D Matrix II.2.java
class Solution {
    public boolean searchMatrix(int[][] matrix, int target) {

        if (matrix.length == 0 || matrix.length == 0) {
            return false;
        }

        int rowNum = matrix.length;
        int colNum = matrix[0].length;

        int left = matrix[0][0];
        int right = matrix[rowNum - 1][r];

        while (left < right) {

            int mid = left + (right - left) / 2;

            if ()
        }

    }
}

Assume you
have an
array of
length n
initialized with all 0's
and are
given k
update operations.

Each operation
is represented
as a triplet:[startIndex,endIndex,inc]
which increments
each element
of subarray A[startIndex...endIndex](startIndex and endIndex inclusive)
with inc.

Return the
modified array
after all
k operations
were executed.

File: mySolutions/0 Amazon_OA/240. Search a 2D Matrix II/240. Search a 2D Matrix II.2.py
class Solution:
    def searchMatrix(self, matrix, target):
        """
        :type matrix: List[List[int]]
        :type target: int
        :rtype: bool
        """
        
        for y in range(len(matrix)):
            find = self.binary_search(matrix[y], target)
            if find:
                return True
            
        return False
    
    def binary_search(self, nums, target):
        
        if len(nums) == 0:
            return False
        
        start = 0
        end = len(nums) - 1
        
        while (start < end):
            mid = start + (end - start) // 2
            if nums[mid] == target:
                return True
            elif nums[mid] < target:
                start = mid + 1
            else:
                end = mid
        return nums[end] == target
                
        